src/nvim/os/fs.c

```diff
@@ -391,6 +391,37 @@ int os_open(const char* path, int flags, int mode)
   return r;
 }
 
+/// Set a file to close on exec.
+//
+// @return -1 if failed to set, 0 otherwise.
+int os_set_cloexec(const int fd)
+{
+#ifdef HAVE_FD_CLOEXEC
+  int e;
+  int fdflags = fcntl(fd, F_GETFD);
+  if (fdflags < 0) {
+    e = errno;
+    ELOG("Failed to get flags on descriptor %d: %s", fd, strerror(errno));
```

Comment:
>  Linux man page says C99 and POSIX.1-2008 require the return value of strerror() to be non-NULL.

Yeah, I didn't know if all targeted OS's conform to at least one of these -- I remembered that there was a small window between supported platforms and posix 2008 compatibility with the `FD_DUP_CLOEXEC` flag (that I mentioned above), but didn't know about `C99` so thought it better to play safe.

Assuming you're confident they all conform, I'm happy -- I couldn't find any pages that didn't.